plugins/scaffolder-react/src/next/components/Stepper/Stepper.tsx

```diff
@@ -163,54 +163,63 @@ export const Stepper = (stepperProps: StepperProps) => {
     });
   }, [steps, activeStep, validators, apiHolder]);
 
-  const handleBack = () => {
+  const handleBack = useCallback(() => {
     setActiveStep(prevActiveStep => prevActiveStep - 1);
-  };
-
-  const handleChange = (e: IChangeEvent) => {
-    setStepsState(current => {
-      const newState = [...current];
-      newState[activeStep] = {
-        ...e.formData,
-      };
-      return newState;
-    });
-  };
-
-  const currentStep = useTransformSchemaToProps(steps[activeStep], { layouts });
+  }, [setActiveStep]);
 
-  const handleNext = async ({
-    formData = {},
-  }: {
-    formData?: Record<string, JsonValue>;
-  }) => {
-    // The validation should never throw, as the validators are wrapped in a try/catch.
-    // This makes it fine to set and unset state without try/catch.
-    setErrors(undefined);
-    setIsValidating(true);
-
-    const returnedValidation = await validation(formData);
-
-    setIsValidating(false);
-
-    if (hasErrors(returnedValidation)) {
-      setErrors(returnedValidation);
-    } else {
+  const handleChange = useCallback(
+    (e: IChangeEvent) => {
       setStepsState(current => {
         const newState = [...current];
         newState[activeStep] = {
-          ...formData,
+          ...e.formData,
```

Comment:
I think if you do something like:

```ts
setStepsState(current => {
  if(isEqual(e.formData, current[activeStep]) {
      return current;
  }
  const newState = ...
  return newState;
});
```

it should allow you to remove the JSON.stringify while creating `formData`:

```ts
const formData = useMemo(
    () => stepsState[activeStep],
    [stepsState, activeStep],
)
```